cg/meta/transfer/external_data.py

```diff
@@ -5,210 +5,204 @@
 from housekeeper.store.models import Version
 
 from cg.apps.slurm.slurm_api import SlurmAPI
-from cg.constants import HK_FASTQ_TAGS
+from cg.constants import HK_FASTQ_TAGS, FileExtensions
+from cg.constants.constants import CaseActions
 from cg.meta.meta import MetaAPI
 from cg.meta.rsync.sbatch import ERROR_RSYNC_FUNCTION, RSYNC_CONTENTS_COMMAND
+from cg.meta.transfer.utils import are_all_fastq_valid
 from cg.models.cg_config import CGConfig
 from cg.models.slurm.sbatch import Sbatch
 from cg.store.models import Case, Customer, Sample
-from cg.utils.checksum.checksum import check_md5sum, extract_md5sum
+from cg.utils.files import get_files_matching_pattern
 
 LOG = logging.getLogger(__name__)
 
 
 class ExternalDataAPI(MetaAPI):
     def __init__(self, config: CGConfig):
         super().__init__(config)
-        self.destination_path: str = config.external.hasta
-        self.source_path: str = config.external.caesar
-        self.base_path: str = config.data_delivery.base_path
         self.account: str = config.data_delivery.account
+        self.base_path: str = config.data_delivery.base_path
+        self.customer_id: str | None = None
+        self.destination_path: str = config.external.hasta
+        self.dry_run: bool | None = None
+        self.force: bool | None = None
         self.mail_user: str = config.data_delivery.mail_user
-        self.slurm_api: SlurmAPI = SlurmAPI()
         self.RSYNC_FILE_POSTFIX: str = "_rsync_external_data"
+        self.slurm_api: SlurmAPI = SlurmAPI()
+        self.source_path: str = config.external.caesar
+        self.ticket: str | None = None
+
+    def _set_parameters(self, ticket: str, dry_run: bool, force: bool | None = False):
+        """Set the parameters for the API."""
+        self.ticket = ticket
+        self.dry_run = dry_run
+        self.force = force
+        self.customer_id = self.status_db.get_customer_id_from_ticket(ticket=self.ticket)
 
-    def create_log_dir(self, dry_run: bool, ticket: str) -> Path:
+    def _create_log_dir(self) -> Path:
         """Creates a directory for log file to be stored"""
         timestamp: dt.datetime = dt.datetime.now()
         timestamp_str: str = timestamp.strftime("%y%m%d_%H_%M_%S_%f")
-        folder_name: Path = Path("_".join([ticket, timestamp_str]))
+        folder_name: Path = Path("_".join([self.ticket, timestamp_str]))
         log_dir: Path = Path(self.base_path, folder_name)
         LOG.info(f"Creating folder: {log_dir}")
-        if dry_run:
+        if self.dry_run:
             LOG.info(f"Would have created path {log_dir}, but this is a dry run")
             return log_dir
         log_dir.mkdir(parents=True, exist_ok=False)
         return log_dir
 
-    def get_source_path(
-        self,
-        customer: str,
-        ticket: str,
-        cust_sample_id: str | None = "",
-    ) -> Path:
+    def _get_source_path(self) -> Path:
```

Comment:
It will be very confusing to me having both a property called `source_path` and a method called `_get_source_path` and these do not give the same value.

Reply:
fixed